packages/stg-evm-v2/devtools/src/credit-messaging/state.ts

```diff
@@ -0,0 +1,138 @@
+import { OmniEdge, OmniGraphBuilder, OmniNode, OmniPoint, OmniVector, isZero } from '@layerzerolabs/devtools'
+import { createConnectedContractFactory, getNetworkNameForEid } from '@layerzerolabs/devtools-evm-hardhat'
+import { createModuleLogger } from '@layerzerolabs/io-devtools'
+import { Options } from '@layerzerolabs/lz-v2-utilities'
+import { OAppEnforcedOption } from '@layerzerolabs/ua-devtools'
+import { createOAppFactory } from '@layerzerolabs/ua-devtools-evm'
+
+import { MessagingAssetConfig } from '../messaging'
+import { getFromCacheOrChain } from '../utils/cache'
+import { decodeEnforcedOptions } from '../utils/enforcedOptions'
+
+import { createCreditMessagingFactory } from './factory'
+import { CreditMessagingEdgeConfig, CreditMessagingNodeConfig, CreditMessagingOmniGraph } from './types'
+
+export async function loadCreditMessagingState(
+    configGraph: CreditMessagingOmniGraph
+): Promise<CreditMessagingOmniGraph> {
+    const logger = createModuleLogger(`State Loader for CreditMessaging`)
+    logger.verbose('Loading CreditMessaging state...')
+
+    const nodePromises: Promise<OmniNode<CreditMessagingNodeConfig>>[] = configGraph.contracts.map(
+        ({ point, config }) => getFromCacheOrChain('CreditMessagingNode', point, () => getNodeFromChain(point, config))
+    )
+
+    const edgePromises: Promise<OmniEdge<CreditMessagingEdgeConfig>>[] = []
+    for (const { vector, config } of configGraph.connections) {
+        edgePromises.push(
+            getFromCacheOrChain('CreditMessagingEdge', vector.from, () => getEdgeFromChain(vector, config)).then(
+                (edge) => {
+                    // Adapt to account for printing bigints
+                    if (edge.config.gasLimit) {
+                        edge.config.gasLimit = BigInt(edge.config.gasLimit)
+                    }
```

Comment:
Why is this necessary? Both the config and the SDK return value should be normalized by schemas to `bigint` so this will only ever be a `bigint`.

If not, this will not work for `0` values as `if (edge.config.gasLimit)` will skip to `else` if the value is `0`